### PR TITLE
ruby examples added

### PR DESCRIPTION
**Pull request by orlando-labs**

### CODE REVIEW

**Comment by fsatka** [comment]
Good morning, thank you for your contribution!

I to go to leave some remarks that will have improved consistency with other examples.

File: ruby/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Intro
+
+Пример использования STT Tinkoff VoiceKit на Ruby
```
Comment (fsatka) [inline ruby/README.md]: Can you translate all section from russian to english, please.

File: ruby/protobuf/protos/google/api/annotations.proto [hunk 2]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015, Google Inc.
```
Comment (fsatka) [inline ruby/protobuf/protos/google/api/annotations.proto]: Can you delete *protobuf* directory and provide proto source from [this dir](https://github.com/TinkoffCreditSystems/voicekit-examples/tree/master/apis/tinkoff/cloud), please.

It is need for regenerate proto on new releasing proto files.

File: ruby/recognize_grpc.rb [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+require_relative 'request'
+require 'json'
+require_relative 'audio'
+
+options = Options.parse
+
+pcm, sample_rate, num_channels, total_frames = Audio.read_frames options
+
+request = RecognitionRequest.new(
+  api_key: ENV['API_KEY'],
```
Comment (fsatka) [inline ruby/recognize_grpc.rb]: This naming of *env* missmatch with naming on [main page](https://github.com/TinkoffCreditSystems/voicekit-examples#setup-environment).

File: ruby/recognize_grpc.rb [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  api_secret: ENV['SECRET_KEY'],
+  audio_frames: pcm,
+  sample_rate: sample_rate,
+  num_channels: num_channels,
+  format: :grpc,
+  options: options
+)
+
+stub = Tinkoff::Cloud::Stt::V1::SpeechToText::Stub.new options[:endpoint] || 'stt.tinkoff.ru:433', GRPC::Core::ChannelCredentials.new
```
Comment (fsatka) [inline ruby/recognize_grpc.rb]: Defoult https port is *443*.

File: ruby/Gemfile.lock [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    googleapis-common-protos-types (1.0.5)
+      google-protobuf (~> 3.11)
+    grpc (1.32.0-universal-darwin)
+      google-protobuf (~> 3.13)
+      googleapis-common-protos-types (~> 1.0)
+    grpc-tools (1.32.0)
+    i18n (1.8.5)
+      concurrent-ruby (~> 1.0)
+    jwt (2.2.2)
+    miniaudio (0.1.0)
```
Comment (fsatka) [inline ruby/Gemfile.lock]: I can't build with this version of miniaudio. I advise you to release the version.

File: ruby/recognize_grpc.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+begin
+  response = stub.recognize request.body, metadata: request.headers
+rescue GRPC::BadStatus => e
+  puts "Error: #{e}"
+  exit 1
+end
+
+puts "Received response:"
+puts JSON.pretty_generate(response)
```
Comment (fsatka) [inline ruby/recognize_grpc.rb]: My running failed on this row with next err:
```
Expected hash arguments. (ArgumentError)
```
I think, that *response* not allowed for *pretty_generate*

File: ruby/README.md (hunk 1 above)
Comment (orlando-labs) [reply ruby/README.md]: Done

File: ruby/recognize_grpc.rb (hunk 3 above)
Comment (orlando-labs) [reply ruby/recognize_grpc.rb]: Sure!

File: ruby/Gemfile.lock (hunk 5 above)
Comment (orlando-labs) [reply ruby/Gemfile.lock]: Promoted to actual v0.3

**Comment by orlando-labs** [comment]
Hi, @fsatka. Thanks for reviewing. There's a bunch of updates pushed and added to PR. Please review and feel free to give any feedback. Thanks.

File: ruby/Gemfile [hunk 7]
```diff
@@ -0,0 +1,8 @@
+ruby '>=2.5.0'
```
Comment (fsatka) [inline ruby/Gemfile:1]: is it necessary add in this file?
```
source 'https://rubygems.org'
```
I can't compile without *source*.
Or can you share another way for compilation, please?

File: ruby/README.md [hunk 8]
```diff
@@ -0,0 +1,43 @@
+# Intro
```
Comment (fsatka) [inline ruby/README.md]: Add *Ruby* in [list](https://github.com/TinkoffCreditSystems/voicekit-examples#language-specific-instructions) with languages examples, please.

File: ruby/protobuf/google/api/annotations.proto [hunk 9]
```diff
@@ -0,0 +1,31 @@
+// Copyright (c) 2015, Google Inc.
```
Comment (fsatka) [inline ruby/protobuf/google/api/annotations.proto]: Delete *google* dir from protobuf, please.
You can get google proto files from [third_party](https://github.com/TinkoffCreditSystems/voicekit-examples/tree/master/third_party) for grpc generation.

The [example](https://github.com/TinkoffCreditSystems/voicekit-examples/blob/master/csharp/generate_protobuf.sh#L18) of *protoc* generation with source from *third_party*

File: ruby/Gemfile (hunk 7 above)
Comment (orlando-labs) [reply ruby/Gemfile:1]: It is normal syntax for bundler Gemfile. 'source' just tells bundler where to find gems. So, yes, it is necessary

File: ruby/README.md (hunk 8 above)
Comment (orlando-labs) [reply ruby/README.md]: Sure, done

File: ruby/protobuf/google/api/annotations.proto (hunk 9 above)
Comment (orlando-labs) [reply ruby/protobuf/google/api/annotations.proto]: Done

File: ruby/Gemfile (hunk 7 above)
Comment (fsatka) [reply ruby/Gemfile:1]: Maybe add it here or represent in README, that it is neccessary.
I think, that code must compile by tutorial without hidden appends in configuration(Gemfile in this case).

File: ruby/README.md [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+    -v, --do-not-perform-vad         Specify this to disable voice activity detection. All audio is processed as though it were a single utterance
+    -s [THR],                        Silence threshold in seconds for VAD to assume the current utterance is ended and the next utterance shall begin.
+        --silence-duration-threshold
+    -p                               Specify this to disable automatic punctuation in recognition results
+        --disable-automatic-punctuation
+    -f, --input-file [FILE]          Path to input file. Supported wav, mp3, flac, oga.
+    -e, --endpoint [ENDPOINT]        Network endpoint
+```
+
+
```
Comment (fsatka) [inline ruby/README.md:43]: Can we describe best way, how to generate proto?
I found somethink like that:
```
rake protobuf:generate
```

File: ruby/recognize_grpc.rb [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+require_relative 'request'
+require 'json'
+require_relative 'audio'
+
+options = Options.parse
+
+pcm, sample_rate, num_channels, total_frames = Audio.read_frames options
+
+request = RecognitionRequest.new(
+  api_key: ENV['VOICEKIT_API_KEY'],
```
Comment (fsatka) [inline ruby/recognize_grpc.rb]: I found, that if voicekit keys not initialized, then occur next error.
```
in `urlsafe_decode64': undefined method `tr' for nil:NilClass (NoMethodError)
```
Can we implement checking on empty keys? And if it empty raise "human readable" exception?

File: ruby/README.md (hunk 10 above)
Comment (orlando-labs) [reply ruby/README.md:43]: Yes, there's a rake task for that. Added to readme

File: ruby/recognize_grpc.rb (hunk 11 above)
Comment (orlando-labs) [reply ruby/recognize_grpc.rb]: OK, just pushed it

**Comment by fsatka** [comment]
Thank you for contribution, we are glad to see new implementations for another methods in future.

Happy new year and merry christmas!

**Pull request merged**